app/controllers/concerns/liquid_enabled_resource.rb

```diff
@@ -0,0 +1,39 @@
+module LiquidEnabledResource
+  extend ActiveSupport::Concern
+
+  included do
+    helper_method :liquid_assigns
+  end
+
+  def liquid_assigns
+    @liquid_assigns ||= default_liquid_assigns.merge!(liquid_resource_assigns)
+  end
+
+  # To be overwritten by each controller
+  def liquid_resource_assigns
+    raise 'Controller included LiquidEnabledResource without defining a liquid_resource_assigns'
```

Comment:
What's the reason for enforcing this overwrite vs just using a `{}` as a default? That way, we won't have to define it on every controller that doesn't use this yet.

Reply:
You want the controller to have a way to define the local assigns, and you want the concern to include that when defined. Either you default to {} in the concern and do a `defined?` call to see if the controller defines a new collection, or you do something like this.

We can also take the HasFields approach:
https://github.com/dradis/dradis-ce/blob/main/app/models/concerns/has_fields.rb#L4-L9

Reply:
I'm with @aapomm on this. I don't like the need for a bunch of mostly empty methods added in every controller. Applying `{}` as the default and overriding it as needed feels better. I don't think that requires the use of a `defined?` check. It's just method overriding.

Even that might be improved by a strict usage instead of method overrides. Something more along the [lines of permissions](https://github.com/dradis/dradis-plugins/blob/main/app/controllers/concerns/dradis/plugins/persistent_permissions.rb#L36-L39) so the assigns get defined as the requirement for the module is added. Although this is just a psudo example and I think instance variables would be hard to access like this.

```
include LiquidEnabledResource

liquid_assigns proc { { 'issue' => IssueDrop.new(@issue) } }
```

Reply:
fixed in 7ff6745b